.circleci/config.yml

```diff
@@ -76,20 +83,27 @@ commands:
       tarball-path:
         type: string
         default: solc-js.tgz
-      install-command:
-        type: string
-        default: npm install
     steps:
       - run:
-          name: Inject solc-js from the tarball into dependencies at <<parameters.path>>
+          name: "Sanity check: tarball exists and the target dir contains a JS project"
           command: |
             [[ -f "<<parameters.tarball-path>>" ]]
+            [[ -f "<<parameters.path>>/package.json" ]]
+      - run:
+          name: Inject solc-js from the tarball into dependencies at <<parameters.path>>
+          command: |
             absolute_tarball_path=$(realpath "<<parameters.tarball-path>>")
-            for solc_module in $(find "<<parameters.path>>" -type d -path "*/node_modules/solc"); do
-                pushd "${solc_module}/../.."
-                <<parameters.install-command>> "$absolute_tarball_path" --ignore-workspace-root-check
-                popd
-            done
+            cd "<<parameters.path>>"
+            mv package.json original-package.json
+            # NOTE: The 'overrides' feature requires npm >= 8.3
+            jq ". + {overrides: {solc: \"${absolute_tarball_path}\"}}" original-package.json > package.json
+            npm install
+      - run:
+          name: "Sanity check: all transitive dependencies successfully replaced with the tarball"
+          command: |
+            solc_version=$(jq --raw-output .version solc-js/package.json)
+            cd "<<parameters.path>>"
+            npm list solc | grep 'solc@' | grep -v "solc@${solc_version}"
```

Comment:
Doing this with grep is not great but a more reliable check will be much longer. `npm list` can give me JSON but it's deeply nested. I tried to create a `jq` command that would handle it but it's getting complicated so I decided to stay with grep for now.